### PR TITLE
session: use the uniform log format for session

### PR DESCRIPTION
**Pull request by jackysp**
### What problem does this PR solve? 
Use the uniform log format for session.

### What is changed and how it works?
Use the uniform log format for session.

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

**Comment by jackysp** [comment]
https://github.com/pingcap/parser/pull/227 merge it first.

File: session/bootstrap.go [hunk 1]
```diff
@@ -462,14 +463,16 @@ func upgrade(s Session) {
...
 		if v >= currentBootstrapVersion {
 			// It is already bootstrapped/upgraded by a higher version TiDB server.
 			return
 		}
-		log.Errorf("[Upgrade] upgrade from %d to %d error", ver, currentBootstrapVersion)
-		log.Fatal(err)
+		log.Error("[Upgrade] upgrade error",
+			zap.Int64("from", ver),
+			zap.Int("to", currentBootstrapVersion))
+		log.Fatal(err.Error())
```
Comment (zimulala) [inline session/bootstrap.go]: Could we put these two logs together？

File: session/txn.go [hunk 2]
```diff
@@ -161,17 +161,17 @@ func mockAutoIDRetry() bool {
...
-			st,
-			st.mutations,
-			st.dirtyTableOP,
-			st.buf,
-			debug.Stack())
+		log.Error("The code should never run here",
+			zap.String("TxnState", st.GoString()),
+			zap.String("mutations", fmt.Sprintf("%#v", st.mutations)),
+			zap.String("dirtyTableOP", fmt.Sprintf("%#v", st.dirtyTableOP)),
+			zap.String("buf", fmt.Sprintf("%#v", st.buf)),
```
Comment (zimulala) [inline session/txn.go]: Could we add the `String()` for `buf`? Maybe other places will use it too.

File: session/bootstrap.go [hunk 3]
```diff
@@ -238,7 +239,7 @@ const (
 func bootstrap(s Session) {
 	b, err := checkBootstrapped(s)
 	if err != nil {
-		log.Fatal(err)
+		log.Fatal(err.Error())
```
Comment (zimulala) [inline session/bootstrap.go]: Do we need to add more information to these logs?

File: session/session.go [hunk 4]
```diff
@@ -267,13 +268,13 @@ func (s *session) StoreQueryFeedback(feedback interface{}) {
 	if s.statsCollector != nil {
 		do, err := GetDomain(s.store)
 		if err != nil {
-			log.Debug("domain not found: ", err)
+			log.Debug("Domain not found", zap.String("error", err.Error()))
```
Comment (winoros) [inline session/session.go]: We can directly use `zap.Error(err)`

File: session/bootstrap.go (hunk 3 above)
Comment (jackysp) [reply session/bootstrap.go]: There're too many of them. It is better to do it in another PR.

**Comment by jackysp** [comment]
PTAL @lysu @zimulala

File: session/session.go [hunk 5]
```diff
@@ -1191,7 +1223,8 @@ func (s *session) Auth(user *auth.UserIdentity, authentication []byte, salt []by
 		}
 	}
 
-	log.Errorf("User connection verification failed %s", user)
+	log.Error("User connection verification failed",
+		zap.String("user", user.String()))
```
Comment (lysu) [inline session/session.go]: `zap.Stringer` can avoid `String()` call if error log level isnt' enable.

File: session/session.go [hunk 6]
```diff
@@ -601,17 +615,25 @@ func (s *session) retry(ctx context.Context, maxCnt uint) (err error) {
 			}
 		}
 		if !s.isRetryableError(err) {
-			log.Warnf("[%s] con:%d session:%v, err:%v in retry", label, connID, s, err)
+			logutil.Logger(ctx).Warn("SQL",
+				zap.String("label", label),
+				zap.String("session", s.String()),
```
Comment (lysu) [inline session/session.go]: zap.Stringer

File: session/session.go [hunk 7]
```diff
@@ -382,7 +384,10 @@ func (s *session) doCommitWithRetry(ctx context.Context) error {
 		// BatchInsert already commit the first batch 1000 rows, then it commit 1000-2000 and retry the statement,
 		// Finally t1 will have more data than t2, with no errors return to user!
 		if s.isRetryableError(err) && !s.sessionVars.BatchInsert && commitRetryLimit > 0 {
-			log.Warnf("[%s] con:%d retryable error: %v, txn: %#v", s.getSQLLabel(), s.sessionVars.ConnectionID, err, &s.txn)
+			logutil.Logger(ctx).Warn("SQL",
+				zap.String("label", s.getSQLLabel()),
+				zap.Error(err),
+				zap.String("txn", s.txn.GoString()))
```
Comment (lysu) [inline session/session.go]: maybe we can make another PR to let zap support `zap.GoStringer` :laughing:

File: session/tidb.go [hunk 8]
```diff
@@ -68,15 +69,19 @@ func (dm *domainMap) Get(store kv.Storage) (d *domain.Domain, err error) {
 	ddlLease = schemaLease
 	statisticLease = statsLease
 	err = util.RunWithRetry(util.DefaultMaxRetries, util.RetryInterval, func() (retry bool, err1 error) {
-		log.Infof("store %v new domain, ddl lease %v, stats lease %d", store.UUID(), ddlLease, statisticLease)
+		log.Info("New domain",
+			zap.String("store", store.UUID()),
+			zap.String("ddl lease", ddlLease.String()),
+			zap.String("stats lease", statisticLease.String()))
```
Comment (lysu) [inline session/tidb.go]: zap.Stringer

File: session/session.go [hunk 9]
```diff
@@ -267,13 +268,13 @@ func (s *session) StoreQueryFeedback(feedback interface{}) {
 	if s.statsCollector != nil {
 		do, err := GetDomain(s.store)
 		if err != nil {
-			log.Debug("domain not found: ", err)
+			logutil.Logger(context.Background()).Debug("Domain not found", zap.Error(err))
```
Comment (zimulala) [inline session/session.go]: s/Domain/domain

File: session/session.go [hunk 10]
```diff
@@ -267,13 +268,13 @@ func (s *session) StoreQueryFeedback(feedback interface{}) {
...
 			metrics.StoreQueryFeedbackCounter.WithLabelValues(metrics.LblError).Inc()
 			return
 		}
 		err = s.statsCollector.StoreQueryFeedback(feedback, do.StatsHandle())
 		if err != nil {
-			log.Debug("store query feedback error: ", err)
+			logutil.Logger(context.Background()).Debug("Store query feedback", zap.Error(err))
```
Comment (zimulala) [inline session/session.go]: s/Store/store

File: session/session.go [hunk 11]
```diff
@@ -408,7 +412,9 @@ func (s *session) doCommitWithRetry(ctx context.Context) error {
 	}
 
 	if err != nil {
-		log.Warnf("con:%d finished txn:%#v, %v", s.sessionVars.ConnectionID, &s.txn, err)
+		logutil.Logger(ctx).Warn("Commit failed",
```
Comment (zimulala) [inline session/session.go]: s/Commit/commit
Please update other places as well.

File: session/session.go [hunk 12]
```diff
@@ -573,10 +579,16 @@ func (s *session) retry(ctx context.Context, maxCnt uint) (err error) {
 			if retryCnt == 0 {
 				// We do not have to log the query every time.
 				// We print the queries at the first try only.
-				log.Warnf("con:%d schema_ver:%d retry_cnt:%d query_num:%d sql:%s%s", connID, schemaVersion, retryCnt,
-					i, sqlForLog(st.OriginText()), sessVars.GetExecuteArgumentsInfo())
+				logutil.Logger(ctx).Warn("Retrying",
+					zap.Int64("schemaVersion", schemaVersion),
+					zap.Uint("retry_cnt", retryCnt),
```
Comment (zimulala) [inline session/session.go]: s/retry_cnt/retryCnt

**Comment by zimulala** [comment]
PTAL @lysu @winoros
